### PR TITLE
Make Helm storage namespace configurable

### PR DESCRIPTION
**Pull request by gtracer**

### CODE REVIEW

**Review by hiddeco — COMMENTED** [review]
This looks like the right approach to me. I have some concerns around forthcoming complaints from users that they have to configure a value twice, but am at the same time of opinion that the Helm storage should in most cases (except when doing out-of-cluster releases) live in the same namespace as the `HelmRelease`.

NB: can you please sign-off your commit (`git commit --amend --signoff`)?

**Comment by stefanprodan** [comment]
@gtracer please run `make test` and commit the changes to pass the e2e tests.

File: api/v2beta1/helmrelease_types.go [hunk 1]
```diff
@@ -69,6 +69,14 @@ type HelmReleaseSpec struct {
 	// +optional
 	TargetNamespace string `json:"targetNamespace,omitempty"`
 
+	// StorageNamespace for helm.
+	// Defaults to the namespace of the HelmRelease.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Optional
+	// +optional
+	StorageNamespace string `json:"storageNamespace,omitempty"`
```
[the remark this answers is not in the document]
Comment (gtracer) [reply api/v2beta1/helmrelease_types.go:78]: Will update

**Review by hiddeco — APPROVED** [review]
Thank you @gtracer :100: 

One last thing I'd like to ask is if you can smash your commits together :-)

**Pull request merged**